addons/account/models/account_payment.py

```diff
@@ -330,7 +330,8 @@ def _compute_partner_bank_id(self):
         for pay in self:
             available_partner_bank_accounts = pay.partner_id.bank_ids.filtered(lambda x: x.company_id in (False, pay.company_id))
             if available_partner_bank_accounts:
-                pay.partner_bank_id = available_partner_bank_accounts[0]._origin
+                if not pay.partner_bank_id:
```

Comment:
You still need to change it if the partner of the bank set is not the right one

Reply:
Indeed, you're right :)